lib/disposable.dart

```diff
@@ -192,29 +215,46 @@ class Disposable implements _Disposable, DisposableManager {
         .then(_completeDisposeFuture);
   }
 
-  /// Automatically dispose another object when this object is disposed.
-  ///
-  /// The parameter may not be `null`.
+  /// Creates a [Timer] instance that will be cancelled if active upon disposal.
+  @mustCallSuper
+  @override
+  Timer getManagedTimer(Duration duration, void callback()) {
+    _InternalDisposable disposable;
+    var timer = new Timer(duration, () {
+      _internalDisposables.remove(disposable);
+      callback();
+    });
+
+    disposable = new _InternalDisposable(() async => timer.cancel());
+    _internalDisposables.add(disposable);
+
+    return timer;
+  }
+
+  /// Creates a periodic [Timer] that will be cancelled if active upon disposal.
+  @mustCallSuper
+  @override
+  Timer getManagedPeriodicTimer(Duration duration, void callback(Timer timer)) {
+    var timer = new Timer.periodic(duration, callback);
```

Comment:
I thought the same thing. Cancelling a timer is idempotent so disposing a cancelled timer will have no effect. The only drawback is that we hold onto the the timer for the life of the Disposable object. I am going to implement `Timer` and add a lifecycle hook for when the timers complete or are cancelled.